### ISSUE TITLE
mkfs for OCF device failed

### ISSUE THREAD

**Issue by jessehui** [spdk/spdk#671]
Please use the issue tracker only for reporting suspected issues.

See [The SPDK Community Page](http://www.spdk.io/community/) for other SPDK communications channels.

Use mkfs for OCF device and failed.

## Expected Behavior

mkfs success.

## Current Behavior

```shell
[root@storage27 mkfs (v19.01.x)]$ ./mkfs /root/Development/SPDK/iscsi.conf.in Cache0
Starting SPDK v19.01 / DPDK 18.11.0 initialization...
[ DPDK EAL parameters: spdk_mkfs --no-shconf -c 0x3 --base-virtaddr=0x200000000000 --file-prefix=spdk_pid236868 ]
EAL: Detected 112 lcore(s)
EAL: Detected 2 NUMA nodes
EAL: No free hugepages reported in hugepages-1048576kB
EAL: Probing VFIO support...
app.c: 625:spdk_app_start: *NOTICE*: Total cores available: 2
reactor.c: 298:_spdk_reactor_run: *NOTICE*: Reactor started on core 0
reactor.c: 298:_spdk_reactor_run: *NOTICE*: Reactor started on core 1
Inserting cache Cache0
Cache0: Successfully added
Cache0: Cache mode : wt
Cache0: Super block config offset : 0 kiB
Cache0: Super block config size : 3576 B
Cache0: Super block runtime offset : 0 kiB
Cache0: Super block runtime size : 1512 B
Cache0: Reserved offset : 0 kiB
Cache0: Reserved size : 128 kiB
Cache0: Core config offset : 0 kiB
Cache0: Core config size : 360 kiB
Cache0: Core runtime offset : 0 kiB
Cache0: Core runtime size : 1172 kiB
Cache0: Core UUID offset : 0 kiB
Cache0: Core UUID size : 16384 kiB
Cache0: Cleaning offset : 0 kiB
Cache0: Cleaning size : 53712 kiB
Cache0: Eviction offset : 0 kiB
Cache0: Eviction size : 35772 kiB
Cache0: Collision offset : 0 kiB
Cache0: Collision size : 53712 kiB
Cache0: List info offset : 0 kiB
Cache0: List info size : 76312 kiB
Cache0: Hash offset : 0 kiB
Cache0: Hash size : 4484 kiB
Cache0: Cache line size: 4 kiB
Cache0: Metadata capacity: 269 MiB
Cache0: OCF metadata self-test PASSED
Cache0: Done saving cache state!
Cache0: Successfully attached
Cache0.core0: Successfully added
Initializing filesystem on bdev Cache0...src/ocf/concurrency/ocf_cache_concurrency.c: 851:_ocf_req_lock_wr_common: *ERROR*: BUG
[1]    236868 abort (core dumped)  ./mkfs /root/Development/SPDK/iscsi.conf.in Cache0

```

## Possible Solution

```c
ENV_BUG_ON(req->map[i].rd_locked);
```
Triggered by this located in ocf/src/concurrency/ocf_cache_concurrency.c

## Steps to Reproduce

1. build spdk with OCF enable
3. config the OCF device: 
```shell
[OCF]
    OCF Cache0 wt Nvme1n1p1 Nvme0n1p1
```
4. run ` ./mkfs /root/Development/SPDK/iscsi.conf.in Cache0`

## Context (Environment including OS version, SPDK version, etc.)

OS: fedora26(4.15)
SPDK: v19.01.x
OCF: v18.12.x

**Comment by jessehui** [comment]
@Donaim Any idea why this is happening?

**Comment by jessehui** [comment]
Put ocf into /usr/src/ocf can pass.

**Comment by Donaim** [comment]
> @Donaim Any idea why this is happening?

Not yet, investigating.

**Comment by jessehui** [comment]
@Donaim Any update on this?

**Comment by Donaim** [comment]
@jessehui Issue is reproducible, top level trigger is known - it's sending discard & write_zeroes during mkfs, but cause is not known. I am also not sure if this is bdev issue or OCF itself. Working on this.

**Comment by Donaim** [comment]
@jessehui the error was fixed by patch f155c98084c718adcbec020d0d976bcf57bccce1, it does not reproduces on current master. We don't know what was the cause yet. Does that help with your issue?

**Comment by jessehui** [comment]
@Donaim Just run the same test and ejected the same error message using master branch with commit ID  `136c3fb46184b4`. 
```shell
[root@storage27 mkfs (master)]$ ./mkfs /root/Development/SPDK/iscsi.conf.in Cache0
Starting SPDK v19.04-pre / DPDK 19.02.0 initialization...
[ DPDK EAL parameters: spdk_mkfs --no-shconf -c 0x3 --log-level=lib.eal:6 --base-virtaddr=0x200000000000 --match-allocations --file-prefix=spdk_pid238956 ]
EAL: No free hugepages reported in hugepages-1048576kB
app.c: 635:spdk_app_start: *NOTICE*: Total cores available: 2
reactor.c: 233:_spdk_reactor_run: *NOTICE*: Reactor started on core 0
reactor.c: 233:_spdk_reactor_run: *NOTICE*: Reactor started on core 1
Inserting cache Cache0
Cache0: Successfully added
Cache0: Cache mode : wt
Cache0: Super block config offset : 0 kiB
Cache0: Super block config size : 3544 B
Cache0: Super block runtime offset : 0 kiB
Cache0: Super block runtime size : 1512 B
Cache0: Reserved offset : 0 kiB
Cache0: Reserved size : 128 kiB
Cache0: Core config offset : 0 kiB
Cache0: Core config size : 360 kiB
Cache0: Core runtime offset : 0 kiB
Cache0: Core runtime size : 1172 kiB
Cache0: Core UUID offset : 0 kiB
Cache0: Core UUID size : 16384 kiB
Cache0: Cleaning offset : 0 kiB
Cache0: Cleaning size : 53712 kiB
Cache0: Eviction offset : 0 kiB
Cache0: Eviction size : 35772 kiB
Cache0: Collision offset : 0 kiB
Cache0: Collision size : 53712 kiB
Cache0: List info offset : 0 kiB
Cache0: List info size : 76312 kiB
Cache0: Hash offset : 0 kiB
Cache0: Hash size : 4484 kiB
Cache0: Cache line size: 4 kiB
Cache0: Metadata capacity: 269 MiB
Cache0: OCF metadata self-test PASSED
Cache0: Done saving cache state!
Cache0: Successfully attached
Cache0.core0: Successfully added
Initializing filesystem on bdev Cache0...src/ocf/concurrency/ocf_cache_concurrency.c: 851:_ocf_req_lock_wr_common: *ERROR*: BUG
[1]    238956 abort (core dumped)  ./mkfs /root/Development/SPDK/iscsi.conf.in Cache0

```

**Comment by Donaim** [comment]
@jessehui we submitted a fix b949f1318ef94a3225bb64b7fd7e7d1ddf47ebc6 that is now on master, could you test if it works for you?

**Comment by jessehui** [comment]
Yep. It works. Thank you.
